l1-contracts/test/staking_asset_handler/addValidator.t.sol

```diff
@@ -0,0 +1,161 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.8.27;
+
+import {StakingAssetHandlerBase} from "./base.t.sol";
+import {StakingAssetHandler, IStakingAssetHandler} from "@aztec/mock/StakingAssetHandler.sol";
+import {IStakingCore} from "@aztec/core/interfaces/IStaking.sol";
+
+// solhint-disable comprehensive-interface
+// solhint-disable func-name-mixedcase
+// solhint-disable ordering
+
+contract AddValidatorTest is StakingAssetHandlerBase {
+  address public canAddValidator = address(0xdead);
+
+  function setUp() public override {
+    super.setUp();
+    stakingAssetHandler = new StakingAssetHandler(
+      address(this),
+      address(stakingAsset),
+      address(staking),
+      WITHDRAWER,
+      MINIMUM_STAKE,
+      10, // mintInterval, usually overridden in test
+      3, // depositsPerMint, usually overridden in test
+      new address[](0)
+    );
+    stakingAsset.addMinter(address(stakingAssetHandler));
+    stakingAssetHandler.grantAddValidatorPermission(canAddValidator);
+  }
+
+  function test_WhenCallerIsNotCanAddValidator(
+    address _caller,
+    address _attester,
+    address _proposer
+  ) external {
+    // it reverts
+    vm.assume(_caller != canAddValidator && _caller != address(this));
+    vm.expectRevert(
+      abi.encodeWithSelector(IStakingAssetHandler.NotCanAddValidator.selector, _caller)
+    );
+    vm.prank(_caller);
+    stakingAssetHandler.addValidator(_attester, _proposer);
+  }
+
+  modifier whenCallerIsCanAddValidator() {
+    // Use the canAddValidator address
+    _;
+  }
+
+  modifier whenItNeedsToMint() {
+    // By default it needs to mint
+    _;
+  }
+
+  function test_WhenNotEnoughTimeHasPassedSinceLastMint(uint256 _interval)
+    external
+    whenCallerIsCanAddValidator
+    whenItNeedsToMint
+  {
+    _interval = bound(_interval, block.timestamp + 1, block.timestamp + 1e18);
+    stakingAssetHandler.setMintInterval(_interval);
+
+    // it reverts
+    vm.expectRevert(
+      abi.encodeWithSelector(IStakingAssetHandler.NotEnoughTimeSinceLastMint.selector, 0, _interval)
+    );
+    vm.prank(canAddValidator);
+    stakingAssetHandler.addValidator(address(0), address(0));
+  }
+
+  modifier whenEnoughTimeHasPassedSinceLastMint(uint256 _interval) {
+    _interval = bound(_interval, block.timestamp + 1, block.timestamp + 1e18);
+    stakingAssetHandler.setMintInterval(_interval);
+    vm.warp(block.timestamp + _interval);
+    _;
+  }
+
+  function test_WhenMaxDepositsPerMintIs0(uint256 _interval, address _attester, address _proposer)
+    external
+    whenCallerIsCanAddValidator
+    whenItNeedsToMint
+    whenEnoughTimeHasPassedSinceLastMint(_interval)
+  {
+    // it reverts
+    stakingAssetHandler.setDepositsPerMint(0);
+
+    vm.assume(_attester != address(0));
+    vm.assume(_proposer != address(0));
+
+    vm.expectRevert(abi.encodeWithSelector(IStakingAssetHandler.CannotMintZeroAmount.selector));
```

Comment:
As mentioned think this one should probably be checked when the value is set, not when it is used.